Javatrix.java

```diff
@@ -117,5 +115,85 @@ public double get(int i, int j) throws ArrayIndexOutOfBoundsException
     {
         return matrix;
     }
+
+    /**
+     * Get column dimension.
+     *
+     * @return n number of columns
+     */
+    public int getColumnDimension()
+    {
+        return n;
+    }
+
+    /**
+     * Get row dimension.
+     *
+     * @return m number of rows
+     */
+    public int getRowDimension()
+    {
+        return m;
+    }
+
+    /**
+     * Copy the internal two-dimensional array.
+     *
+     * @return matrixCopy copy of internal array
+     */
+    public double[][] getArrayCopy()
+    {
+        double[][] matrixCopy = new double[m][n];
+        for (int i = 0; i < m; i++)
+        {
+            for (int j = 0; j < n; j++)
+            {
+                matrixCopy[i][j] = matrix[i][j];
+            }
+        }
+        return matrixCopy;
+    }
```

Comment:
This line is from getArrayCopy, which I implemented just before constructWithCopy. The getArrayCopy function is supposed to return the 2D array. This pull request is actually for constructWithCopy, which does return a Javatrix object. Sorry for the confusion - this is probably what Dr Fenwick was referring to today when he discussed submitting small code changes for reviews!